src/ground_main.c

```diff
@@ -48,19 +49,434 @@ extern void nullsub_120();
+extern void IncrementPlayTime(struct PlayTimeStruct *);
+extern void WaitForNextFrameAndAdvanceRNG(void);
+extern void LoadBufferedInputs(void);
+extern void nullsub_120(void);
+extern void sub_80A5E70(void);
+extern void sub_809B638(void);
+extern void nullsub_106(void);
+extern void sub_80A73EC(void);
+extern void sub_8099750(void);
+extern void sub_8009908(void);
+extern void GroundMap_Reset(void);
+extern void sub_809D0BC(void);
+extern void DeleteGroundEvents(void);
+extern void DeleteGroundLives(void);
+extern void DeleteGroundObjects(void);
+extern void DeleteGroundEffects(void);
+extern void sub_809C658(void);
+extern void nullsub_16(void);
+extern void UpdateAdventureAchievements(void);
+extern void xxx_call_update_bg_sound_input(void);
+extern void sub_80A6E68(void);
+extern void sub_80060EC(void);
+extern void sub_801180C(void);
+extern void ScriptPrintNullTextbox(void);
+extern void sub_809977C(void);
+extern void GroundMap_ExecuteEvent(s32, u32);
+extern void UpdateFadeInTile(s32);
+extern void GroundMap_Action(void);
+extern void sub_8098C58(void);
+extern void sub_8001D88(void);
+extern u32 GroundMainGameCancelRequest(u32 r0);
+extern void sub_80999D4(s32);
+extern void sub_809D4B0(void);
+extern void sub_809C63C(void);
+extern void sub_8005838(s32, s32);
+
+u32 xxx_script_related_8098468(s32 param_1)
+{
+    s32 r7;
+    s32 varE;
+
+    gUnknown_20398B4 = param_1;
+    gUnknown_20398B9 = 0;
+    gUnknown_20398B8 = 1;
+    gScriptMode = 0;
+
+    switch (param_1) {
+        case 0xd:
+            gScriptMode = 1;
+            gUnknown_20398B8 = 0;
+            gUnknown_20398B9 = 1;
+            break;
+        case 0xe:
+            gUnknown_20398B8 = 1;
+            gUnknown_20398B9 = 1;
+            break;
+        case 0xf:
+            gUnknown_20398B8 = 1;
+            gUnknown_20398B9 = 1;
+            break;
+        case 0x10: {
+            s32 local_1c;
+            u32 auStack24;
+
+            GetScriptVarScenario(3,&local_1c,&auStack24);
+            gUnknown_20398B8 = local_1c == 0;
+            gUnknown_20398B9 = 1;
+            break;
+        }
+        case 0x11:
+            gUnknown_20398B8 = 0;
+            gUnknown_20398B9 = 1;
+            break;
+        default:
+            break;
+    }
+    sub_801180C();
+    if (gUnknown_20398B9 == 0 && !sub_80023E4(0xd)) {
+        FadeOutAllMusic(0x10);
+    }
+    gUnknown_2026E4E = 0x808;
+    UpdateFadeInTile(0);
+    sub_8099648();
+    sub_809975C();
+    sub_809D0AC();
+    sub_80A7744();
+    sub_809D490();
+    sub_80A62F0();
+    sub_809C5C4();
+    sub_809A560();
+    sub_809CB50();
+    AllocGroundMapAction();
+    nullsub_117();
+    AllocGroundEvents();
+    AllocGroundLives();
+    AllocGroundObjects();
+    AllocGroundEffects();
+    sub_809A62C();
+    gUnknown_203B49C = 0;
+    gUnknown_203B49D = 0;
+    gUnknown_20398A8 = 1;
+    gUnknown_20398AC = 0;
+    gUnknown_20398B0 = -1;
+    gUnknown_20398BE = GetScriptVarValue(0,0xd);
```

Comment:
Can use the defines from `include/event_flag.h` to document all the ScriptVar func calls in this file.

Reply:
Good idea, added the defines